src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/adjudications/AdjudicationService.kt

```diff
@@ -483,15 +471,9 @@ class AdjudicationService(
       ?: throw BadDataException("Prison internal location $internalLocationId not found")
   }
 
-  private fun findBooking(prisoner: Offender): OffenderBooking {
-    return prisoner.bookings.firstOrNull { it.bookingSequence == 1 }
-      ?: throw BadDataException("Prisoner ${prisoner.nomsId} has no bookings")
-  }
-
-  private fun Offender.findLatestBooking(): OffenderBooking {
-    return this.bookings.firstOrNull { it.bookingSequence == 1 }
-      ?: throw BadDataException("Prisoner ${this.nomsId} has no bookings")
-  }
+  private fun findLatestBooking(offenderNo: String): OffenderBooking =
+    offenderBookingRepository.findLatestByOffenderNomsId(offenderNo)
+      ?: throw NotFoundException("Prisoner $offenderNo not found or has no bookings")
```

Comment:
The downside of getting the latest booking in one shot is that we cant tell whether the prisoner doesnt exist or it has no bookings, but we probably dont care

Reply:
agreed